### PR TITLE
Delete finished jobs

### LINKED ISSUE #344: Delete finished jobs
We could offer the option to automatically delete jobs from the DB as soon as they're finished. Im guessing that for many use cases, this would be much more useful than running a periodic job to delete them later.

(We could also imagine a hooks system where we could provide builtin actions or let users define their custom actions)

### PR DESCRIPTION
**Pull request by mxd4**
Closes #344

This PR adds a new parameter when creating a worker, allowing the deletion of jobs on completion,
either for successful jobs or all jobs.

### Successful PR Checklist:

- [x] Tests
  - [ ] (not applicable?)
- [x] Documentation
  - [ ] (not applicable?)
- [x] Had a good time contributing?

### CODE REVIEW

**Comment by ewjoachim** [comment]
Woo, I'm excited to see the rest. The start is looking awesome !

File: procrastinate/cli.py
Comment (ewjoachim) [inline procrastinate/cli.py:172]: Good job on that ! I'm impressed :)

File: procrastinate/periodic.py
Comment (ewjoachim) [inline procrastinate/periodic.py:75]: Haha oops :sweat_smile:

File: tests/integration/test_manager.py
Comment (ewjoachim) [inline tests/integration/test_manager.py:250]: I think this test was actually meaningless, even before your PR. We should have a look at it, in another PR.

File: tests/migration/test_migration.py
Comment (ewjoachim) [inline tests/migration/test_migration.py:16]: So just to have it explicitely said, this allows the tests to be launched from a subfolder.

File: tests/unit/test_worker.py
Comment (ewjoachim) [inline tests/unit/test_worker.py:117]: Those 2 new tests are A-OK !

**Review by ewjoachim — APPROVED** [review]
This looks amazing :) Thank you so much for your contribution !

**Comment by elemoine** [comment]
This looks really good, I just want to make sure that we do the right thing w.r.t. migrations. I'd like to have a closer look at that part specifically before we merge.

File: procrastinate/sql/schema.sql
Comment (elemoine) [inline procrastinate/sql/schema.sql:208]: (nit) s/where/WHERE

File: procrastinate/sql/migrations/00.17.00_02_delete_finished_jobs.sql
Comment (elemoine) [inline procrastinate/sql/migrations/00.17.00_02_delete_finished_jobs.sql:6]: (nit) s/where/WHERE

**Review by elemoine — APPROVED** [review]
I made two minor comments that you may want to take into account before merging. Otherwise looks great to me. I paid attention to the migration part, and that looks good to me too! Merge away!

**Pull request merged**
